media/js/src/GridAsset.jsx

```diff
@@ -99,8 +100,12 @@ export default class GridAsset extends React.Component {
     onSelectedAnnotationUpdate(annotation) {
         const type = this.asset.getType();
         const a = this.props.asset.annotations[annotation];
+        let label = `asset: ${this.props.asset.title}`;
 
-        this.setState({selectedAnnotation: a});
+        if (a) {
+            label = `annotation: ${a.title}`;
+        }
+        this.setState({selectedAnnotation: a, label: label});
```

Comment:
I just realized, this `label` state variable is unnecessary - it's just derived off the `selectedAnnotation` state. It's better to handle this in the `render` step, to simplify state changes. State variables should only be used when absolutely necessary. I'll add a commit here to do this refactor.

Reply:
Yeah I tried this originally and kept getting an error when I would click back to the original asset for some reason... Not doing it now though